### ISSUE TITLE
[feature] Payment Gateway handling fees

### ISSUE THREAD

**Issue by anondev-sudo** [Ctrlpanel-gg/panel#351]
# 🚀 Feature Request

### Is your proposal related to a problem?

*Provide a clear and concise description of what the problem is.
For example, "I can't change x and it makes me have to do manual work"*

I use tax to cover payment gateway provider fees. Paypal and Stripe don't have the same fees so it's unfair to have the same tax for both gateways. 

### Describe the solution you'd like

*Provide a clear and concise description of what you want to happen.
For Example, "Automate the changing of x so I don't have to do it manually"*

Add config option in admin area to set different tax for every payment gateway.

**Better Alternative:**
It would be nice it option to set handling fee in % per payment gateway would be added in configuration tab. The handling should be added as second item to invoice (on checkout page, payment gateway and on pdf invoice)

### Additional context

*Is there anything else you can add about the proposal?
You might want to link to related issues here, if you haven't already.*

https://github.com/ControlPanel-gg/dashboard/issues/290#issuecomment-1001627783

**Comment by IceToast** [comment]
You mean a fee set for a specific payment gateway correct? 
This has nothing to do with taxes, as taxes have nothing to do with the payment gateway but the country you are handling your payments in

**Comment by anondev-sudo** [comment]
I want to set different tax in % for Paypal and for Stripe. I use tax to cover payment gateway fees as currenctly there is no option to add Payment Gateway handling in %.

**Comment by IceToast** [comment]
Did you overread my comment maybe?

Why would you add a TAX for a specific payment gateway? Taxes are regulated by law at your country.
What you are talking about is a fee for a payment gateway (the cut paypal or stripe takes from each payment).

**Comment by anondev-sudo** [comment]
I use tax to cover payment gateway fees as currenctly **there is no option to set Payment Gateway handling fee** in %.


Add option to set handling fee in % per Payment Gateway.

**Comment by IceToast** [comment]
> I use tax to cover payment gateway fees as currenctly **there is no option to set Payment Gateway handling fee** in %.
> 
> Add option to set handling fee in % per Payment Gateway.

Yes, thats exactly what i said in my first comment. It's not a tax it's a fee. Thats much different ^^

Anyways i would like to ask you to be a little more polite to the contributors at cp.gg. It's not motivating to have someone say "Do this, do that".

**Comment by anondev-sudo** [comment]
I wanted per payment gateway tax setting because my suggestion to add handling fee option was closed here https://github.com/ControlPanel-gg/dashboard/issues/290#issuecomment-1001627783



> most likely wont do that. **just add this procentage to the tax**, adjust your product prices or deal with the fees^^ <3

I was told here to use tax for this purpose.

**Comment by 1day2die** [comment]
Because for now we have different priorities. If a contributor decides to code that we can reopen that issue

**Comment by IceToast** [comment]
> I wanted per payment gateway tax setting because my suggestion to add handling fee option was closed here [#290 (comment)](https://github.com/ControlPanel-gg/dashboard/issues/290#issuecomment-1001627783)
> 
> > most likely wont do that. **just add this procentage to the tax**, adjust your product prices or deal with the fees^^ <3
> 
> I was told here to use tax for this purpose.

Your issue #290 was about adding a custom item to the checkout which can be interpreted as user customizable. Which is much different to a fixed (percentage can be set by user) item that does not need to be customized in any way by the value...

**Comment by anondev-sudo** [comment]
Maybe there could be just option on the payment gateways settings to set custom fixed fee + fee in % per gateway (as not all gateways have the same fees). It should be displayed as Payment Gateway handling fee on the invoice/order confirmation. Or there could be option to add custom item to checkout (also with settings to set it only for some gateway(s), product(s), the item price could be in % - based on the current order amount or fixed price), this would be the most customisable  to serve all needs of users.

**Comment by IceToast** [comment]
The first description would fixes the problem the best. That's also how i would implement it.

**Comment by S0ly** [comment]
@1day2die have this been done ?

**Comment by Neustradamus** [comment]
Any progress on it?

**Comment by MrWeez** [comment]
Implemented and will be available in next release
